tests/api/enqueue.cpp

```diff
@@ -439,3 +439,147 @@ TEST_F(WithCommandQueue, FinishAfterFlush) {
         thread->join();
     }
 }
+
+#ifdef CLVK_UNIT_TESTING_ENABLED
+TEST_F(WithCommandQueue, EnqueueTooManyCommands) {
+
+    static const unsigned NUM_INSTANCES =
+        CLVK_CONFIG_GET(max_entry_points_instances);
+
+    static const char* program_source = R"(
+    kernel void test_simple(global uint* out, uint id)
+    {
+        out[id] = id;
+    }
+    )";
+
+    auto cfg_force_descriptor_set_allocation_failure =
+        CLVK_CONFIG_SCOPED_OVERRIDE(force_descriptor_set_allocation_failure,
+                                    bool, true, true);
+    CLVK_CONFIG_ASSERT_EQ(descriptor_sets_allocate_retry_sleep_us, UINT32_MAX);
+    // We need to make sure that the kernel don't start executing before we
+    // enqueue one too many
+    CLVK_CONFIG_ASSERT_GT(max_first_cmd_batch_size, NUM_INSTANCES);
+
+    // Create kernel
+    auto kernel = CreateKernel(program_source, "test_simple");
+
+    // Create buffer
+    size_t buffer_size = NUM_INSTANCES * sizeof(cl_uint);
+    auto buffer = CreateBuffer(CL_MEM_WRITE_ONLY | CL_MEM_ALLOC_HOST_PTR,
+                               buffer_size, nullptr);
+
+    // Dispatch kernel
+    size_t gws = 1;
+    size_t lws = 1;
+
+    SetKernelArg(kernel, 0, buffer);
+    cl_uint i;
+    for (i = 0; i < NUM_INSTANCES - 1; i++) {
+        SetKernelArg(kernel, 1, &i);
+        EnqueueNDRangeKernel(kernel, 1, nullptr, &gws, &lws);
+    }
+    cl_event event;
+    SetKernelArg(kernel, 1, &i);
+    EnqueueNDRangeKernel(kernel, 1, nullptr, &gws, &lws, 0, nullptr, &event);
+    i++;
+    Flush();
+
+    SetKernelArg(kernel, 1, &i);
+    cl_uint err;
+    cl_uint count = 0;
+    do {
+        err = clEnqueueNDRangeKernel(m_queue, kernel, 1, nullptr, &gws, &lws, 0,
+                                     nullptr, nullptr);
+        count++;
+    } while (err == CL_OUT_OF_RESOURCES);
+    cl_int event_cmd_execution_status;
+    GetEventInfo(event, CL_EVENT_COMMAND_EXECUTION_STATUS,
+                 &event_cmd_execution_status);
+    ASSERT_EQ(event_cmd_execution_status, CL_COMPLETE);
+    ASSERT_GT(count, 10);
```

Comment:
Where does the 10 come from?

Reply:
I'm not sure what I was thinking here, I'll simplify this test.